src/handlers/proposalsHandlers.ts

```diff
@@ -56,6 +56,44 @@ const getProposals = (
     });
 };
 
+const getProposal = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void => {
+  db.sql('proposals.selectById', { id: req.params.id })
+    .then((proposalsQueryResult: Result<Proposal>) => {
+      logger.debug(proposalsQueryResult);
+      if (proposalsQueryResult.row_count === 0) {
+        res.status(404)
+          .contentType('application/json')
+          .send({ message: 'Not found. Find existing proposals by calling with no parameters.' });
+        return;
+      }
+      const proposal = proposalsQueryResult.rows[0];
+      if (isProposal(proposal)) {
+        res.status(200)
+          .contentType('application/json')
+          .send(proposal);
+      } else {
+        next(new InternalValidationError(
+          'The database responded with an unexpected format.',
+          isProposal.errors ?? [],
```

Comment:
Is `isProposal.errors` correct?

I've glanced through the code and see this pattern repeatedly, but I'm a little mystified how it works. `isProposal` is a schema validation method, I take it, so how does calling `.errors` on it return errors specific to the `isProposal(proposal)` check that was done on [L74](https://github.com/PhilanthropyDataCommons/service/pull/192/files#diff-77b7da863212124379e28ebf124a964abd60d0e05f56a0d7fdb0a077a1061e22R74)?

Reply:
It is the AJV way, I suppose. https://ajv.js.org/api.html#validation-errors says this:
> In case of validation failure, Ajv assigns the array of errors to errors property of validation function

I agree it's kind of odd. It might be a source of trouble for multiple calls, is that what you are seeing?

Reply:
Holy moly, that's wild! Clearly you're correct and this is the AJV way so I'll mark this resolved, but what a bananas design IMO. Presumably there are some performance benefits they're gaining from it.

Reply:
Looking further down it says:
> In case of asynchronous validation, the returned promise is rejected with exception Ajv.ValidationError that has errors property.

So perhaps they really expect us to use async validation because that seems better. Speculating.